solution/backend/resources/views/resources.py

```diff
@@ -54,59 +60,84 @@ def get_annotated_url(self):
             default=None,
         )
 
-    def get_gov_results(self, query, page):
-        key = 'M1igE4Qcfo8LLQr7o_I9KLA6qkybmlC9IRhVCCbFbl4='
-        offset = (page - 1) * self.limit
-        results = []
-        rstring = f'https://search.usa.gov/api/v2/search/?affiliate=reg-pilot-cms-test&access_key={key}' \
-                  f'&query={query}&limit={self.limit}&offset={offset}'
-        gov_results = json.loads(requests.get(rstring).text)
+    def format_gov_results(self, gov_results):
         if "errors" in gov_results or (gov_results["web"]["total"] == 0 and not gov_results["web"]["results"]):
             raise NotFound("Invalid page")
-
+        results = []
         for gov_result in gov_results['web']['results']:
-            result = {
+            results.append({
                 'name': gov_result['title'],
                 'snippet': gov_result['snippet'],
                 'url': gov_result['url']
-            }
-            results.append(result)
+            })
         self.gov_results = {
             "total": gov_results["web"]["total"],
             "results": results,
         }
 
+    def get_gov_results(self, query, page):
+        key = 'M1igE4Qcfo8LLQr7o_I9KLA6qkybmlC9IRhVCCbFbl4='
+        offset = (page - 1) * self.limit
+
+        rstring = f'https://search.usa.gov/api/v2/search/?affiliate=reg-pilot-cms-test&access_key={key}' \
+                  f'&query={urlparse.quote_plus(query)}&limit={self.limit}&offset={offset}'
```

Comment:
we need to encode it again on the backend. Django decoded the query from the front end which would work but since we have to make an external search it gets chopped off on the way to search.gov